### ISSUE TITLE
unexpected genfrac error "bad math style"

### ISSUE THREAD

**Issue by pkra** [mathjax/MathJax#3068]
Fairly minimal example would be $$\genfrac{}{}{-1pt}{4}{a}{b}$$. 

On both mathjax.org#demo and our almost-v4-alpha build, this gives an error of "Bad math style for \genfrac". 

However, "real" TeX seems ok with this.

**Comment by pkra** [comment]
Full example: `\begin{equation} \lim _{x\to -\infty }H_{\delta }(x) = 0\text{ and }\lim _{x\genfrac {}{}{-1pt}{4}{\to }{&lt;} {\mathcal{F}}_{\mathrm{god}}^{(q)}(a,b)} H_{\delta }(x)= +\infty . \cssId{texmlid38}{\tag{17}} \end{equation}`

**Comment by dpvc** [comment]
Well, this is a very strange usage of `\genfrac` that is outside its documented functionality.  First, the 4th argument should be `0`, `1`, `2`, or `3`, so there is no defined meaning for `4` in that position.  It is that issue that MathJax is complaining about in the error message, and I would say the error message is correct.  The expression is taking advantage of quirks of the definition of `\genfrac`; I looked up the actual definition of `\genfrac` to see what it does in that case, and due to the structure of an "if" statement, anything other than `0`, `1`, or `2`, is treated as a `3` (for `\scriptscriptstyle`).  While that *is* how actual LaTeX works, we have not always tried to model LaTeX's behavior for what is in reality an invalid argument.  It would not be hard in this case to make any other value be treated as `3`, but you should be aware that this expression is not really valid and is taking advantage of the quirks of the definition.

Similarly, the third argument is supposed to be the thickness of the rule for the fraction, with `0pt` being no rule.  It is not clear what a negative value should do in this case, and I could not find any documentation for negative values.  Because this ends up relying on the primitive `\above` control sequence (rather than a macro), whose definition is in the actual Pascal code for TeX (or its translation into C), I didn't try to look up what a negative number does, though looking at the actual boxes that TeX produces in this case, it ends up creating a `\rule` with that negative height and 0 depth.  The effect seems to be to remove the spacing between the boxes for the numerator and denominator (both of which have the standard height and depth of the font).  The size of the negative amount doesn't seem to matter (the size of the kerns above and below the rule with negative height are adjusted to compensate for the negative height).  So it seems that this is an undocumented effect of a negative value for the line thickness.

Because the kerns are adjusted in response to the negative value, it seems that in this case removing the spacing between the numerator and denominator is the intended result, so that might want to be implemented in MathJax (currently, this case is very badly handled).  But again, this is still taking advantage of an undocumented feature.

I will look into making a PR to address the issues
